pwndbg/commands/__init__.py

```diff
@@ -301,19 +310,19 @@ def _OnlyWithResolvedHeapSyms(*a, **kw):
 sudo dpkg --add-architecture i386
 sudo apt-get install libc-dbg:i386
 """
-                    % function.__name__
-                )
-                print(
-                    message.warn(
-                        "pwndbg can still try to use this command without debug symbols by `set resolve-heap-via-heuristic on`."
                     )
-                )
-                print(message.warn("You can show your current config about heap by `heap_config`."))
-                print(
-                    message.warn(
+                    w(
+                        "pwndbg can still try to use this command without debug symbols by `set resolve-heap-via-heuristic on`.\n"
+                        "You can show your current config about heap by `heap_config`.\n"
                         "Then pwndbg will resolve some missing symbols via heuristics, but the results of those commands may be incorrect in some cases."
                     )
+            elif pwndbg.glibc.get_version() is None:
+                e("Can't resolve the heap since the GLIBC version is not set.")
+                w(
+                    "Please set the GLIBC version you think the target binary was compiled (using `set glibc <version>` command; e.g. 2.32) and re-run this command."
                 )
+            else:
+                e("An unknown error occurred when resolved the heap.")
```

Comment:
Shall we tell the user we couldn't find proper symbol addresses and they can set them?

Reply:
Yes, I'll add the hint for it also.